adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/VertxBasedAmqpProtocolAdapter.java

```diff
@@ -674,6 +689,7 @@ protected void onCommandReceived(final ProtonSender sender, final CommandContext
         });
         final Map<String, Object> items = new HashMap<>(4);
         items.put(Fields.EVENT, "command sent to device");
+        metrics.incrementCommandDeliveredToDevice(command.getTenant());
```

Comment:
IMHO this needs to be moved to the block handling the the *accepted* outcome.

Reply:
The _accepted_ outcome is validated with _if/else_ blocks as below.

 `if (Accepted.class.isInstance(remoteState)) {
...
}`. 

I understood that the _incrementCommandDeliveredToDevice_ part should be moved there in consideration with the outcome. Is my understanding right? If not can you please explain where exactly do you mean?